topics/wallet-interaction-standard.md

```diff
@@ -0,0 +1,254 @@
+
+`networks`: A list of networks on which the relying party plans to operate.
+- `chainId`(`text`): The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`scopes`: A list of permission scopes the relying party requires. Possible values: 
+- `"canister_call"`
+
+`challenge` (`blob`): A challenge used for the wallet to sign in order to prove its access to the identity. The challenge should be an array of 32 cryptographically random bytes generated from a secure random source by the sender of the request.
+
+#### Response
+
+`version` (`text`): The version of the standard used.
+
+`appMetadata`: Information about the wallet.
+- `name` (`text`): The user-friendly name of the wallet.
+- `icon` (`text`, optional): An optional URL poiting to an icon resource representing the wallet.
+
+`networks`: A list of networks on which the user has agreed the relying party can operate. This should be a subset of the `networks` from the original request.
+- `chainId` (`text`): The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`scopes`: A list of permission scopes that the user has agreed the relying party can be granted. This should be a subset of the `scopes` field from the original request. Possible values:
+- `"canister_call"`
+
+`identity`: The identity the user has selected.
+- `publicKey` (`blob`): The DER-encoded public key associated with the identity. The public key can be used to [derive a self-authenticating principal](https://internetcomputer.org/docs/current/references/ic-interface-spec/#principal).
+- `ledger` (optional): Ledger data.
+    - `subaccount` (`blob`, optional)
+
+`signature` (`blob`): The signature produced by signing the challenge with the private key associated with the `identity`.
+
+#### Use-Case
+
+1. The relying party sends a `permission` request to the wallet.
+2. Upon receiving the message, the wallet presents the details of the to-be-established connection to the user and asks the user to select the identity that will be paired in response.
+    - If the user approves the request, the wallet sends a succesful response back to the relying party.
+    - If the user rejects the request, the wallet sends a response with an error back to the relying party.
+3. After receiving a succesful response, the relying party verifies that the wallet has access to the private key associated with the provided identity:
+    - The relying party retrieves the `publicKey` from `identity`, determines the `signature` scheme and verifies whether it was generated by signing the `challenge` from the request with the private key associated with the `publicKey`.
```

Comment:
I just realized, we also need to specify a domain separator (see [here](https://internetcomputer.org/docs/current/references/ic-interface-spec/#signatures)), to make sure that the signature cannot be misused in other contexts.

My suggestion would be something like `\x0Aic-wallet-challenge` to be consistent with other domain separators specified in the interface spec (examples in this [section](https://internetcomputer.org/docs/current/references/ic-interface-spec/#authentication)).

Reply:
Good point 👍 I've updated that part